analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SpecifyTimeoutOnRegex.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+class Compliant
+{
+    void Ctor()
+    {
+        var defaultOrder = new Regex("some pattern", RegexOptions.None, TimeSpan.FromSeconds(1)); // Compliant
+
+        var namedArgs = new Regex(
+            pattern: "some pattern",
+            matchTimeout: TimeSpan.FromSeconds(1), // Compliant
+            options: RegexOptions.None);
+    }
+
+    void Instance()
+    {
+        var regex = new Regex("some pattern", RegexOptions.None, TimeSpan.FromSeconds(1));
```

Comment:
Add an example with an implicit object creation (this may require a separate test file with a language version):
```cs
Regex regex = new("some pattern", RegexOptions.None, TimeSpan.FromSeconds(1));
```